File: 0020-valid-parentheses/0020-valid-parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Dictionary<char,char> bracketPairs = new  Dictionary<char,char>();
        bracketPairs.Add('}','{');
        bracketPairs.Add(')','(');
        bracketPairs.Add(']','[');
        
        bool valid = true;
        
        List<char> brackets = new List<char>();
        foreach (char letter in s){
            if(letter == '{' || letter == '(' || letter == '['){
                brackets.Add(letter);
            }
            if(letter == '}' || letter == ')' || letter == ']'){
                if(brackets.Count>0 && bracketPairs[letter] == brackets[brackets.Count-1])
                    brackets.RemoveAt(brackets.Count-1);
                else
                    return false;
            }
            
        
        }
        if (brackets.Count != 0){
            return false;
        }
        return true;
        
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracket_stack = []
        pairs = {
            "(": ")",
            "[": "]",
            "{":"}"
        }

        for ch in s:
            if ch in pairs:
                bracket_stack.append(ch)
            else:
                if bracket_stack:
                    prev = bracket_stack.pop()
                    if ch != pairs[prev]:
                        return False
                else:
                    return False
        
        return True if not bracket_stack else False
